src/foam/u2/grid/GridCell_old.js

```diff
@@ -0,0 +1,347 @@
+                var M = foam.mlang.Expressions.create();
+                if (currMatchId === null || currMatchId === undefined){
+                    return M.NOT(M.HAS(currProperty));
+                }else if (currProperty.cls_.name == "Date"){
+                   return Query.util.inDay(currProperty, currMatchId);
+                } else {
+                    if (matchCurrId){
+                        var p = currProperty.clone();
+                        var pname = currProperty.name; 
+                        p["f"] = function (o) {
+                            var obj = o[pname];
+                            return obj?obj["id"]:obj; 
+                            };
+                        return M.EQ(p, currMatchId);
+                    }else {
+                        return M.EQ(currProperty, currMatchId);
+                    }
+                }
+        }, 
+        
+        function makeCell(){
+            //var M = foam.mlang.Expressions.create();
+            //var pred = M.AND(this.colPredicate, this.rowPredicate);
+
+            if(this.predicate && this.data){
+                var d = this.data.where(this.predicate);
+                if (this.order){
+                    d = d.orderBy(this.order);
+                }
+                d.select().
+                then(function(result){
+                    var div = foam.u2.Element.create("div");
+                    console.log("CELL: row:" + this.rowMatchId + " col:" + this.colMatchId + ", " + result.a.length);
+                    if (! result || !result.a || !result.a.length){
+                        console.log("no result found");
+                    }
+                    var a = result.a;
+                    for (var i=0; i<a.length; i++){
+                        var entry = a[i];
+                        var v = this.getEntryView(entry); 
+                        v.sub("CELL_ENTRY_SELECTED", this.onEntrySelection)
+                        div.add(v); 
+                    }
+                    div.add(this.cellBottom); 
+                    this.cell = div;
+                }.bind(this));
+            }else {
+                var p = foam.u2.Element.create("div");
+                p.add("---");
+                this.cell = p;
+            }
+        },
+        
+        function getEntryView(entry){
+            console.log(entry);
+            var v = this.getCellView(entry);
+            if (self.entrySelectable){
+                v.sub("SELECTED", function(e){
+                    self.selectedEntry = e.src.data; 
+                    self.pub("ENTRY_SELECTION");
+                    }.bind(this));
+            }
+            return v; 
+        }, 
+        
+        function getCellView(a){
+            if (this.cellView){
+                var v = this.cellView$cls.create({of: this.of, data: a});
+                return v; 
+            }
+            var d = foam.u2.Element.create("div");
```

Comment:
U2 Elements support programming in a "fluent" style, which means that this could have been written:

return foam.u2.Element.create("div").
  start('p').add(a.name).end().
  start('p').add(a.lastSeenAlive).end();